base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -372,6 +373,11 @@ async fn check_health(
                     }
                 }
             }
+            if let Ok(Some(mut conn)) = comms.get_connection(result.peer.clone()).await {
+                if let Err(err) = conn.disconnect(Minimized::No).await {
+                    warn!(target: LOG_TARGET, "Failed to disconnect peer {} ({})", result.peer, err);
+                }
+            }
```

Comment:
This connection should not be dropped here, as there are other processes running to get rid of unused outbound connections. 

An incoming connection on a seed peer from this enquiring node is zero cost to the seed node after this conversation is finished, but the connection may have been established previously as a legitimate outbound connection. 

Another consideration is that a consecutive discovery request will still be a proper discovery, although it should be faster, thus coming back to this code while the outbound connection is still open does not invalidate the tests.

Reply:
Disagree. These connections are primary to the seed peers.  The reason our nodes use upwards of 1000 connections is because we keep them open because we might need them later. 
And it's super important to not keep connections open to the seed peers. You are not suppose to use them as a connection hub. If another process wants to use the connection, they can reopen the connection.

Reply:
If we insist on closing the connection, it should only be done if a new connection was established for this purpose, otherwise, a legitimate connection established by another process will be closed.

Reply:
this has the added benefit of doing this as well: #6907 
I don't think this is a big issue to disconnect from the seed peers. They will reopen connections if needed. They are there to seed peers, not seed the network.

Reply:
What if the connection we are closing here is from a custom seed peer we are syncing from? Those should not be disconnected. The code here does not differentiate between DNS seeds and custom seeds. DNS seed connections may be closed, but not custom seeds that are not DNS seeds (one can still add a DNS seed as a custom seed).